algorithms/merging/merge.py

```diff
@@ -216,15 +216,22 @@ def merge_and_truncate(params, experiments, reflections):
         logger.debug(out.getvalue())
 
     if params.reporting.wilson_stats:
-        if not intensities.space_group().is_centric():
-            wilson_scaling = data_statistics.wilson_scaling(
-                miller_array=intensities, n_residues=params.n_residues
-            )  # XXX default n_residues?
-            # Divert output through logger - do with StringIO rather than
-            # info_handle else get way too much whitespace in output.
-            out = StringIO()
-            wilson_scaling.show(out=out)
-            logger.info(out.getvalue())
+        try:
+            if not intensities.space_group().is_centric():
+                wilson_scaling = data_statistics.wilson_scaling(
+                    miller_array=intensities, n_residues=params.n_residues
+                )  # XXX default n_residues?
+                # Divert output through logger - do with StringIO rather than
+                # info_handle else get way too much whitespace in output.
+                out = StringIO()
+                wilson_scaling.show(out=out)
+                logger.info(out.getvalue())
+        except RuntimeError:
```

Comment:
RuntimeError is very broad. Do you know whether the error is raised in the `.wilson_scaling()` or the `.show()` method, and is it possible to not catch errors in the other?

Reply:
Yes, the RuntimeError is raised in `.wilson_scaling`, which I take to be off limits, since it's in `mmtbx`.  Specifically it's a `RuntimeError: mean() argument is an empty array`, occurring at https://github.com/cctbx/cctbx_project/blob/8f1ca5d1c172015c24030314f9599d4b5934680a/mmtbx/scaling/absolute_scaling.py#L798.  Obviously, that should be a `ValueError` or almost anything other than a `RuntimeError`, but I don't have any control over that.

But you're right that I should be keeping the try/except tighter around the problem call.